File: client/python/mesh/prsim/datahouse.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#

from abc import ABC, abstractmethod
from typing import Any, List

from mesh.kinds import Document, Paging, Page
from mesh.macro import mpi, spi


@spi("mesh")
class DataHouse(ABC):

    @abstractmethod
    @mpi("mesh.dh.writes")
    def writes(self, docs: List[Document]):
        """
        batch write log message
        :param docs:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.dh.write")
    def write(self, doc: Document):
        """
        write log message
        :param doc:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.dh.read")
    def read(self, index: Paging) -> Page[Any]:
        """
        write log message
        :param index:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.dh.indies")
    def indies(self, index: Paging) -> Page[Any]:
        """
        Export index list.
        :param index:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.dh.tables")
    def tables(self, index: Paging) -> Page[Any]:
        """
        Export table list.
        :param index:
        :return:
        """
        pass

File: client/java/src/main/java/io/be/mesh/prsim/DataHouse.java
package io.be.mesh.prsim;

import io.be.mesh.macro.MPI;
import io.be.mesh.macro.SPI;
import io.be.mesh.struct.Document;
import io.be.mesh.struct.Page;
import io.be.mesh.struct.Paging;

import java.util.List;

/**
 * @author jianyue.li
 */
@SPI("mesh")
public interface DataHouse {

    /**
     * 批量写入，主要使用
     */
    @MPI(value = "mesh.dh.writes", timeout = 3000, retries = 1)
    void writes(List<Document> docs);

    /**
     * 单条写入
     */
    @MPI(value = "mesh.dh.write", timeout = 3000, retries = 1)
    void write(Document doc);

    /**
     * 单条写入
     */
    @MPI("mesh.dh.read")
    Page<Object> read(Paging index);

    /**
     * Index list.
     */
    @MPI("mesh.dh.indies")
    Page<Object> indies(Paging index);

    /**
     * Talbe list.
     */
    @MPI("mesh.dh.tables")
    Page<Object> tables(Paging index);
}
